### PR TITLE
[expo] Fix dev loading view insets

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Fixes ENG-4718

# How

The native module for react-native-safe-area-context changed and so I updated how we reach into it to pull out the initial insets.

# Test Plan

- Load NCL on an iPhone with a notch / home indicator and make a change to see that the dev menu appears underneath the home indicator
- Apply this patch and notice it now is properly positioned to account for the home indicator

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo/src/environment/DevLoadingView.tsx [hunk 1]
```diff
@@ -101,11 +101,9 @@ export default function DevLoadingView() {
 /**
  * This is a hack to get the safe area insets without explicitly depending on react-native-safe-area-context.
  * The following code is lifted from: https://git.io/Jzk4k
- *
- * TODO: This will need to be updated for Fabric/TurboModules.
  **/
-const RNCSafeAreaProviderConfig = UIManager.getViewManagerConfig('RNCSafeAreaProvider') as any;
-const initialWindowMetrics = RNCSafeAreaProviderConfig?.Constants?.initialWindowMetrics as
+const RNCSafeAreaContext = NativeModules.RNCSafeAreaContext;
```
Comment (ide) [inline packages/expo/src/environment/DevLoadingView.tsx:105]: Intentional not to use turbomodules?
```tsx
const RNCSafeAreaContext = TurboModuleRegistry.get('RNCSafeAreaContext');
RNCSafeAreaContext?.getConstants().initialWindowMetrics;
```

Comment (brentvatne) [reply packages/expo/src/environment/DevLoadingView.tsx:105]: no good reason outside of old habits

Comment (ide) [reply packages/expo/src/environment/DevLoadingView.tsx:105]: Probably better to use the new interface then, which I assume is more likely to last?

Comment (tsapeta) [reply packages/expo/src/environment/DevLoadingView.tsx:105]: `TurboModuleRegistry.get` does support legacy bridge modules when remote debugger is on, so yeah, that sounds like a better solution

File: packages/expo/src/environment/DevLoadingView.tsx [hunk 2]
```diff
@@ -100,14 +100,11 @@ export default function DevLoadingView() {
...
- * TODO: This will need to be updated for Fabric/TurboModules.
  **/
-const RNCSafeAreaProviderConfig = UIManager.getViewManagerConfig('RNCSafeAreaProvider') as any;
-const initialWindowMetrics = RNCSafeAreaProviderConfig?.Constants?.initialWindowMetrics as
-  | { insets: { bottom: number } }
-  | undefined;
+const RNCSafeAreaContext = TurboModuleRegistry.get('RNCSafeAreaContext');
+
+// @ts-ignore: we're not using the spec so the return type of getConstants() is {}
+const initialWindowMetrics = RNCSafeAreaContext?.getConstants?.().initialWindowMetrics;
```
Comment (ide) [inline packages/expo/src/environment/DevLoadingView.tsx:107]: If RNCSafeAreaContext is defined, RNCSafeAreaContext.getConstants() will always be defined so the extra ?. isn't needed:

```suggestion
const initialWindowMetrics = RNCSafeAreaContext?.getConstants().initialWindowMetrics;
```

**Review by ide — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**